addon/components/horizontal-bar-chart.js

```diff
@@ -306,7 +311,15 @@ export default ChartComponent.extend(FloatingTooltipMixin,
       .text((d) => this.get('formatLabelFunction')(d.value))
       .attr(this.get('valueLabelAttrs'));
 
-    var labelWidth = this.get('labelWidth');
+    var labelWidth;
+    // If the chart contains a mix of negative and positive values, the axis
+    // and labels are in the middle of the chart, not at the edges of the chart,
+    // so allow more space to compute how the label is trimmed.
+    if (this.get('hasNegativeValues') && this.get('hasPositiveValues')) {
+      labelWidth = this.get('outerWidth') / 2;
+    } else {
+      labelWidth = this.get('labelWidth');
+    }
```

Comment:
It affects the size of the bars

On Monday, November 23, 2015, akhileshgupta <notifications@github.com>
wrote:

> In addon/components/horizontal-bar-chart.js
> <https://github.com/Addepar/ember-charts/pull/131#discussion_r45700582>:
>
> > @@ -306,7 +311,15 @@ export default ChartComponent.extend(FloatingTooltipMixin,
> >        .text((d) => this.get('formatLabelFunction')(d.value))
> >        .attr(this.get('valueLabelAttrs'));
> >
> > -    var labelWidth = this.get('labelWidth');
> > +    var labelWidth;
> > +    // If the chart contains a mix of negative and positive values, the axis
> > +    // and labels are in the middle of the chart, not at the edges of the chart,
> > +    // so allow more space to compute how the label is trimmed.
> > +    if (this.get('hasNegativeValues') && this.get('hasPositiveValues')) {
> > +      labelWidth = this.get('outerWidth') / 2;
> > +    } else {
> > +      labelWidth = this.get('labelWidth');
> > +    }
>
> @philpee2 <https://github.com/philpee2> after this change, the "Label
> Width" control will have no effect on the charts with both negative and
> positive values. That could be confusing to the end user.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Addepar/ember-charts/pull/131/files#r45700582>.
>


-- 

Phil Nachum
Software Engineer | Addepar <https://addepar.com/>
(917) 834-1367


1215 Terra Bella Ave.
Mountain View, CA 94043